src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs

```diff
@@ -0,0 +1,522 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.Setup;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted
+{
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public class SqlBulkCopyTruncation : IClassFixture<SQLSetupStrategyCertStoreProvider>, IDisposable
+    {
+        private SQLSetupStrategyCertStoreProvider _fixture;
+
+        private readonly Dictionary<string, string> tableNames = new Dictionary<string, string>();
+
+        public SqlBulkCopyTruncation(SQLSetupStrategyCertStoreProvider fixture)
+        {
+            this._fixture = fixture;
+            this.tableNames = fixture.sqlBulkTruncationTableNames;
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsEnclaveEnabledAndConnectionStringsProvided))]
+        [ClassData(typeof(AEConnectionStringProvider))]
+        public void BulkCopyTestsInt(string connectionString)
+        {
+            PopulateTable("TabIntSource", "1, 300", connectionString);
+
+            Assert.Throws<InvalidOperationException>(() => DoBulkCopy(tableNames["TabIntSource"], tableNames["TabTinyIntTarget"], connectionString));
+
+            //Truncate removes the data but not the table.
```

Comment:
I see this pattern of cleaning up/truncating the tables at the end of the test. But if the test fails, the tables will not be truncated. It looks like you need to put the cleanup in a finally block.

Reply:
Add to the applicable parts.